homeassistant/components/aseko_pool_live/config_flow.py

```diff
@@ -60,21 +65,74 @@ async def async_step_user(
                 _LOGGER.exception("Unexpected exception")
                 errors["base"] = "unknown"
             else:
-                await self.async_set_unique_id(info[CONF_UNIQUE_ID])
-                self._abort_if_unique_id_configured()
+                return await self.async_store_credentials(info)
 
-                return self.async_create_entry(
-                    title=info[CONF_EMAIL],
-                    data={CONF_ACCESS_TOKEN: info[CONF_ACCESS_TOKEN]},
+        return self.async_show_form(
+            step_id="user",
+            data_schema=self.data_schema,
+            errors=errors,
+        )
+
+    async def async_store_credentials(self, info: dict[str, Any]) -> ConfigFlowResult:
+        """Store validated credentials."""
+
+        if self.reauth_entry:
+            self.hass.config_entries.async_update_entry(
```

Comment:
We have this helper:

https://github.com/home-assistant/core/blob/e041c3aa0f724311b0a5a83c1c0fa88acd2b4f72/homeassistant/config_entries.py#L2271-L2282